src/app/core/entity-details/related-entities/related-entities.component.ts

```diff
@@ -96,4 +126,27 @@ export class RelatedEntitiesComponent<E extends Entity> implements OnInit {
       return rec;
     };
   }
+
+  private updateColumnsToDisplayForScreenSize() {
+    if (!this.columns) {
+      return;
+    }
+
+    this.columnsToDisplay = this.columns
+      .filter((c) => {
+        const column = toFormFieldConfig(
+          (this.columns ?? []).find((x) => toFormFieldConfig(x).id === c),
+        );
+        if (column?.hideFromTable) {
+          return false;
+        }
+
+        const numericValue = ScreenSize[column?.visibleFrom];
+        if (numericValue === undefined) {
+          return true;
+        }
+        return this.screenWidthObserver.currentScreenSize() >= numericValue;
+      })
+      .map((c) => toFormFieldConfig(c).id);
```

Comment:
Does `toFormFieldConfig` really need to be called 3 times in this function? I think running this once in the beginning would make the code much simpler.

Reply:
I have no clue how this completely pointless .find ended up here. Simplified now :sweat_smile: